megamek/src/megamek/common/weapons/BayWeaponHandler.java

```diff
@@ -542,10 +543,14 @@ public boolean handleAeroSanity(IGame.Phase phase, Vector<Report> vPhaseReport)
                     WeaponAttackAction bayWaa = new WeaponAttackAction(waa.getEntityId(), waa.getTargetType(), waa.getTargetId(), wId);
                     AttackHandler bayWHandler = ((Weapon)bayWType).getCorrectHandler(autoHit, bayWaa, game, server);
                     bayWHandler.setAnnouncedEntityFiring(false);
-                    // This should always be true. Maybe there's a better way to write this?
+                    // This should always be true
                     if (bayWHandler instanceof WeaponHandler) {
                         WeaponHandler wHandler = (WeaponHandler) bayWHandler;
                         wHandler.setParentBayHandler(this);
+                    } else {
+                        logDebug(METHOD_NAME,
+                                "bayWHandler is not a weapon handler! How did you manage that?");
```

Comment:
could you include the entity ID or any other applicable identifier to help locate this problem if it occurs

Reply:
Fixed